client/galaxy/scripts/mvc/collection/list-collection-creator.js

```diff
@@ -278,9 +278,9 @@ var ListCollectionCreator = Backbone.View.extend(BASE_MVC.LoggableMixin)
 
         /** describe what is wrong with a particular element if anything */
         _isElementInvalid: function(element) {
-            // if (element.history_content_type !== "dataset") {
-            //     return _l("is not a dataset");
-            // }
+            if (element.history_content_type !== "dataset" && element.type !== "file") {
+                return _l("is not a dataset or a library file");
+            }
```

Comment:
Well I liked it better before you did this - there is no reason in my mind for this extra check.

Reply:
Really? I mean the check is kinda silly indeed, but I thought it serves a purpose of disallowing collections in?

Reply:
But collections should be allowed in ideally - that used to be allowed - but good point the current version of these wouldn't work with collections.

Reply:
Yeah, they should be allowed in, but this is not what is stopping them as you point out.